docs/en/llm/api_server_reasoning.md

```diff
@@ -0,0 +1,109 @@
+# Reasoning Outputs
+
+For models that support reasoning capabilities, such as [DeepSeek R1](https://huggingface.co/deepseek-ai/DeepSeek-R1), LMDeploy supports parsing the reasoning results in the service and separately records the reasoning content using `reasoning_content`.
+
+## Examples
+
+### DeepSeek R1
+
+We can start the DeepSeek R1 model's api_server service just like launching other models. The difference is that we need to specify `--enable-reasoning` and `--reasoning-parser` parameters.
+In the `--reasoning-parser` parameter, we need to specify the specific parser to use.
+
+```
+lmdeploy serve api_server deepseek-ai/DeepSeek-R1-Distill-Qwen-1.5B \
+    --enable-reasoning --reasoning-parser deepseek_r1
+```
+
+Then, we can call the service's functionality from the client:
+
+```python
+from openai import OpenAI
+
+openai_api_key = "Your API key"
+openai_api_base = "http://0.0.0.0:23333/v1"
+
+client = OpenAI(
+    api_key=openai_api_key,
+    base_url=openai_api_base,
+)
+
+models = client.models.list()
+model = models.data[0].id
+
+messages = [{"role": "user", "content": "9.11 and 9.8, which is greater?"}]
+response = client.chat.completions.create(model=model, messages=messages, stream=True)
+for stream_response in response:
+    print('reasoning content: ',stream_response.choices[0].delta.reasoning_content)
```

Comment:
When running the example, from output messages, it seems the `reasoning_content` is always `None`, is it normal?
```
...
reasoning content:  None
content:  \
reasoning content:  None
content:  ]
reasoning content:  None
content:  
reasoning_content: None
content: First, I need to compare the two numbers, 9.11 and 9.8, to determine which one is greater.
....
```

Reply:
It is at the very beginning of the streaming response.